src/server/handlers/thirdpartyRequests/transactions/{ID}/authorizations.ts

```diff
@@ -0,0 +1,151 @@
+import { Consent } from '../../../../../model/consent'
+import { Scope } from '../../../../../model/scope'
+import { consentDB, scopeDB } from '../../../../../lib/db'
+import { NotFoundError } from '../../../../../model/errors'
+import { verifySignature } from '../../../../../lib/challenge'
+import { Request, ResponseToolkit, ResponseObject } from '@hapi/hapi'
+import {
+  AuthPayload,
+  hasNullFields,
+  hasCorrectStatus,
+  hasActiveConsentKey,
+  hasMatchingScope
+} from '../../../../domain/thirdpartyRequests/transactions/{ID}/authorizations'
+
+/*
+ * The HTTP request `POST /thirdpartyRequests/transactions/{ID}/authorizations`
+ * is used to check auhtorization of the PISP for a transaction identified
+ * by {ID}.
+ * It verifies the user's signature on the quote using the associated Consent's
+ * public key. It is used by the Switch to verify transactions.
+ * The response is sent using outgoing request
+ * `PUT /thirdpartyRequests/transactions/{ID}/authorizations`.
+ */
+export async function post (
+  request: Request, h: ResponseToolkit): Promise<ResponseObject> {
+  // TODO: request validation for headers, source and
+  // payload structure (non existent/extra fields)
+  // TODO: use JOI for these 2 validations?
+  // Is request validation done internally?
+
+  const payload: AuthPayload = request.payload as AuthPayload
+
+  // TODO: use JOI for these 2 validations?
+  // Validate against payload null fields
+  if (hasNullFields(payload)) {
+    return h.response().code(Enum.Http.ReturnCodes.BADREQUEST.CODE)
+  }
+
+  // Validate incoming status
+  if (!hasCorrectStatus(payload)) {
+    return h.response().code(Enum.Http.ReturnCodes.BADREQUEST.CODE)
+  }
+
+  let consent: Consent
+
+  // Check if consent exists and retrieve consent data
+  try {
+    consent = await consentDB.retrieve(payload.consentId)
+  } catch (error) {
+    Logger.push(error)
+    Logger.error('Could not retrieve consent')
+
+    if (error instanceof NotFoundError) {
+      return h.response().code(Enum.Http.ReturnCodes.NOTFOUND.CODE)
+    }
+
+    return h.response().code(Enum.Http.ReturnCodes.BADREQUEST.CODE)
+  }
+
+  let consentScopes: Scope[]
+
+  // Retrieve scopes for the consent
+  try {
+    consentScopes = await scopeDB.retrieveAll(payload.consentId)
+  } catch (error) {
+    Logger.push(error)
+    Logger.error('Could not retrieve scope')
+
+    if (error instanceof NotFoundError) {
+      return h.response().code(Enum.Http.ReturnCodes.NOTFOUND.CODE)
+    }
+
+    return h.response().code(Enum.Http.ReturnCodes.BADREQUEST.CODE)
+  }
+
+  // Check if the request scope matches with the consent
+  if (!hasMatchingScope(consentScopes, payload)) {
+    return h.response().code(Enum.Http.ReturnCodes.NOTFOUND.CODE)
+  }
+
+  // Check for presence of an active key
+  if (!hasActiveConsentKey(consent)) {
+    //
+    // TODO: Is this the correct error code for key not existing? or just 400
+    //
+    return h.response().code(Enum.Http.ReturnCodes.NOTFOUND.CODE)
```

Comment:
Not Found should be used when resource isn't found - but here we have invalid parameter in request so
```suggestion
    return h.response().code(Enum.Http.ReturnCodes.BADREQUEST.CODE)
```